### PR TITLE
Metadata Display

### PR DESCRIPTION
**Pull request by hblitza**
Introduces "Display Metadata" button in layer tree dropdown menu.  

- fetches geonetwork csw over SHOGun metadata/getRecordByUuid.action (@weskamm )
- parses csw json (taken from GFZ, @annarieger, @hwbllmnn  https://gitlab.intranet.terrestris.de/gfz/gfz_shogun_webapp/-/blob/master/src/main/webapp/map-catalog/src/util/MetadataParser.tsx#L29)
- displays the data in simple antd modal (<[Descriptions](https://ant.design/components/descriptions/)>)

@terrestris/devs please review :pray: 

image

### CODE REVIEW

File: src/component/Modal/Metadata/Metadata.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      .then(response => response.json())
+      .then(json => {
+        const metaData = MetadataParser.parseCswRecord(json);
+        this.setState({
+          metadata: metaData
+        });
+      })
+      .catch(error => {
+        // eslint-disable-next-line no-console
+        console.error(error);
```
Comment (weskamm) [inline src/component/Modal/Metadata/Metadata.tsx:62]: please remove this line and the one above

**Comment by hblitza** [comment]
Thanks, let's try it then.

File: src/component/Modal/Metadata/Metadata.tsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  componentDidMount() {
+    this.fetchMetadata(this.props.layer.get('metadataIdentifier'));
+  }
+
+  /**
+   * Fetch metadata by uuid from GNOS
+   * @param layer
+   */
+  fetchMetadata(uuid: string) {
+    const url = '/globewq-webapp/metadata/getRecordByUuid.action?uuid=' + uuid + '&outputFormat=json';
```
Comment (marcjansen) [inline src/component/Modal/Metadata/Metadata.tsx:51]: The above URL looks not very portable to my eyes. How can we have this more generic?

File: src/component/Modal/Metadata/Metadata.tsx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+            {metadata.extent ? <Descriptions.Item label="Geographical Extent:" span={3}>
+              minX: {metadata.extent.minX['#text']}
+              <br />
+              minY: {metadata.extent.minY['#text']}
+              <br />
+              maxX: {metadata.extent.maxX['#text']}
+              <br />
+              maxy: {metadata.extent.maxY['#text']}
+            </Descriptions.Item> : null }
+          </Descriptions>
```
Comment (marcjansen) [inline src/component/Modal/Metadata/Metadata.tsx:128]: None of the above labels are translatable, should they be?

File: src/util/MetadataParser.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        src: ObjectUtil.getValue('gmd:RS_Identifier', cswRec)?.['gmd:code']?.['gco:CharacterString'],
+      },
+      featureTypeName: ObjectUtil.getValue('gmd:MD_Identifier', cswRec)?.['gmd:code']?.['gco:CharacterString'],
+      featureTypeAttribution: ObjectUtil.getValue('gmd:otherCitationDetails', cswRec)?.['gco:CharacterString'],
+      capabilitiesUrl: ObjectUtil.getValue('srv:SV_OperationMetadata', cswRec)?.['srv:connectPoint']?.['gmd:CI_OnlineResource']?.['gmd:linkage']?.['gmd:URL'],
+      graphicOverviewUrl: ObjectUtil.getValue('gmd:MD_DataIdentification', cswRec)?.['gmd:graphicOverview']?.['gmd:MD_BrowseGraphic']?.['gmd:fileName']?.['gco:CharacterString'],
+      credit: ObjectUtil.getValue('gmd:MD_DataIdentification', cswRec)?.['gmd:credit']?.['gco:CharacterString'],
+      supplementalInformation: ObjectUtil.getValue('gmd:MD_DataIdentification', cswRec)?.['gmd:supplementalInformation']?.['gco:CharacterString'],
+      serviceType: ObjectUtil.getValue('srv:SV_ServiceIdentification', cswRec)?.['srv:serviceType']
+    };
```
Comment (marcjansen) [inline src/util/MetadataParser.tsx:64]: This looks as if could be quite well tested. If possible please add them in afollow-up PR.

**Review by marcjansen — COMMENTED** [review]
Three comments, all of which should be tackled in follow-ups.

File: src/component/Modal/Metadata/Metadata.tsx (hunk 2 above)
Comment (weskamm) [reply src/component/Modal/Metadata/Metadata.tsx:51]: whoops nice catch, yes this needs to be fixed

Comment (weskamm) [reply src/component/Modal/Metadata/Metadata.tsx:51]: fixed in https://github.com/terrestris/react-geo-baseclient/pull/654

File: src/component/Modal/Metadata/Metadata.tsx (hunk 3 above)
Comment (weskamm) [reply src/component/Modal/Metadata/Metadata.tsx:128]: fixed in https://github.com/terrestris/react-geo-baseclient/pull/654

File: src/util/MetadataParser.tsx (hunk 4 above)
Comment (hblitza) [reply src/util/MetadataParser.tsx:64]: Okay, I put it in my TODO list. Thanks!

**Pull request merged**